activiti-cloud-services-runtime-bundle/activiti-cloud-services-events/src/main/java/org/activiti/cloud/services/events/listeners/ProcessEngineEventsAggregator.java

```diff
@@ -105,6 +110,20 @@ protected ExecutionContext resolveExecutionContext(CommandContext commandContext
        
     }
     
+    protected void mayBeAddRootExecutionContext(CommandContext commandContext, ExecutionEntity executionEntity) {
+        ExecutionContext rootExecutionContext = commandContext.getGenericAttribute(ROOT_EXECUTION_CONTEXT);
+        
+        if(rootExecutionContext == null && executionEntity.getRootProcessInstanceId() != null) {
+            ExecutionEntity rootProcessInstance = commandContext.getExecutionEntityManager()
+                                                                .findById(executionEntity.getRootProcessInstanceId());
+            
+            rootExecutionContext = createExecutionContext(rootProcessInstance);
+
+            commandContext.addAttribute("rootExecutionContext",
```

Comment:
Literal `rootExecutionContext` could be replaced by constant `ROOT_EXECUTION_CONTEXT `.